Add release notes for 1.4 (#274)
okteto/docs: src/content/self-hosted/install/releases.mdx

id: releases
---

## 1.4.0
<TODO: insert release date here once it's published>

> ### Features

Reviewer: is kubernetes 1.24 supported as part of this release?

Author: Not yet. We'd like to have more certainty before publishing it

> - Update the Destroy All UI 

Reviewer: The feature here is that the logic of `destroy` commands is always executed when a namespace or preview is deleted

Reviewer: @rberrelleza could you help ^^?

Author: Maybe something like 

"Deleting a namespace or preview will inherit the behavior from the `destroy` command"

- Support custom resources (requests and limits) for installer jobs

> - Render External Resource notes using Markdown

Reviewer: I would document a single feature, "external resources", explaining what they are and how to use it.
@jLopezbarb @viddo do we have the field `external` documented for the okteto manifest?

Author: The url reference should work once we merge @jLopezbarb 's pr

- Store External Resource in CRD

> - Get oauth claims from ID Token to be compatible with more providers

Reviewer: Let's mention this is for our Open ID integration

- Support for "resource" and "audience" open id configuration
- Pod requests won't be modified by default by okteto
- Support "virtual services" to show endpoints in the Okteto UI

> - Follow the standards for CRDs: dedicated folder, global option in chart and keep them in helm uninstall by default

Reviewer: Can we add a link where this is documented?

> - Support for custom affinities for dev containers and installer jobs in helm config

Reviewer: Can we add a link where this is documented?

- Add confirmation on sleep action in the UI
- Add external endpoints to preview view
- Add force/retry buttons and namespace logs on namespace delete-failed
- Merge user-provided configuration snippet with the okteto configuration snippet

> - Reorganize fields affecting "nodes" in the "autoscaler" section for better clarity

Reviewer: Can we add a link where this is documented?

- Add search filter for Git Catalog items in the Deploy dialog
- Destroying previews will now also execute the commands from the okteto manifest 
- Add namespace/preview name to the notification when deleted

> - Parametrize the image we use for the initContainer of the daemon

Reviewer: Link to the docs?

> - Parametrize the autoscaler image

Reviewer: Link to the docs?

### Bug Fix

- Redeploy or deletion of a dev environment deployed from a catalog item which was deleted used to display an error to the user
- Don't sleep ingresses not managed by Okteto
- Add tolerations and node selector to telemetry cronjob

> - GitHub App integration repos improvements

Reviewer: Can we explain what's the improvement?

> - Fix Github installations selector infinite loop in autopreviews

Reviewer: This is not available in self-hosted, I wouldn't document it

> - Fix autopreviews ability to manage your own repository

Reviewer: Same thing

id: releases
---

## 1.4.0

> <TODO: insert release date here once it's published>

Reviewer 2: ```suggestion
January 19, 2023
```

id: releases
---

## 1.4.0
<TODO: insert release date here once it's published>

> ### Features

Reviewer 2: Lots of features, that's very cool! Can we sort them so that related features are together?

Author: Reordered!

- Update the Destroy All UI 
- Support custom resources (requests and limits) for installer jobs
- [External Resources](/docs/reference/manifest/#external-object-optional): Now you can use external resources in your okteto manifest. They are a list of external resources to deploy with your development environment like databases, message brokers and lambda functions.

> - Get OpenId claims from ID Token to be compatible with more providers

Reviewer 2: Let's put this as a bug fix since it's not adding new behaviors for customers.

> - Support for "resource" and "audience" OpenId configuration

Reviewer 2: @pchico83 what is the user-facing benefit of this?

> - Pod requests won't be modified by default by okteto

Reviewer 2: Is this a breaking change? IF so, can we move it to the corresponding section?

```suggestion
- By default, pod requests are no longer modified by okteto
```

> - Support "virtual services" to show endpoints in the Okteto UI

Reviewer 2: ```suggestion
- `virtual services` are now displayed as part of the development environment
```

> - Follow the standards for [CRDs](/docs/reference/manifest/#crds): dedicated folder, global option in chart and keep them in helm uninstall by default

Reviewer 2: I think this should be a bugfix. 

```suggestion
- The chart now follows best practices for [CRDs](/docs/reference/manifest/#crds): dedicated folder, global option in the chart, and keep them in helm uninstall by default
```

- Support for custom [affinities](/docs/reference/manifest/#affinity) for dev containers and installer jobs in helm config

> - Add confirmation on sleep action in the UI

Reviewer 2: ```suggestion
- Sleeping a namespace using the UI requires confirmation.
```

> - Add external endpoints to preview view

Reviewer 2: How is this different from external resources?

Reviewer 3: It's actually part of the [aforementioned external resources on line 15](https://github.com/okteto/docs/pull/274/files#diff-1767c76eb78f49fc2e9e695befc095442be9cc21631851c3672e66e80baf6aa3R15), I think we could remove this?

```suggestion
```

> - Add force/retry buttons and namespace logs on namespace delete-failed

Reviewer 2: Is this accurate? not sure what `delete-failed` is. 
```suggestion
- Option to force and retry namespace deletion.
```

Author: Yes, it's like that :+1

> - Merge user-provided configuration snippet with the okteto configuration snippet

Reviewer 2: Whatis this one about? do we have docs or a sample somewhere?

Author: I think it's NGINX config. @pchico83 could you confirm?

Reviewer: Yes, it's the NGINX configuration snippet, done via the annotation `"nginx.ingress.kubernetes.io/configuration-snippet"`.
Before this change, we were overwriting the configuration snippet provided by the user. Now, we merge the user configuration with our configuration

> - Add search filter for Git Catalog items in the Deploy dialog

Reviewer 2: ```suggestion
- Add a search filter to the Catalog in the Deploy dialog
```

- Destroying previews will now also execute the commands from the okteto manifest 

> - Add namespace/preview name to the notification when deleted

Reviewer 2: To which notification?

Author: Rephrased it! It's the notification of deleting the space

> - Parametrize the image we use for the initContainer of the [daemon](/docs/reference/manifest/daemonset)

Reviewer 2: ```suggestion
- The initContainer image of the [daemon](/docs/reference/manifest/daemonset) can be overridden in the Helm configuration file.
```

> - Parametrize the [autoscaler](/docs/reference/manifest/autoscaler) image and reorganize fields affecting "nodes" for better clarity

Reviewer 2: ```suggestion
- The container image [autoscaler](/docs/reference/manifest/autoscaler) can be overridden in the Helm configuration file.
```

> - Parametrize the [autoscaler](/docs/reference/manifest/autoscaler) image and reorganize fields affecting "nodes" for better clarity

Reviewer 2: Is the reorganization a breaking change?

Reviewer: No, the old syntax also works

### Bug Fix

> - Redeploy or deletion of a dev environment deployed from a catalog item which was deleted used to display an error to the user

Reviewer: - User creates a catalog item **C**
- User deploys the catalog item **C** into a dev environment **D**
- User deletes the catalog item **C**
- If user re-deploys the dev environment **D** was getting an error because catalog item **C** no longer exists

- Don't sleep ingresses not managed by Okteto
- Add tolerations and node selector to telemetry cronjob

> - GitHub App integration repos improvements: Display private repositories added from the user in the GitHub App.

Author: Private repositories (added by the user to the github app) weren't displaying in the github tab integration for deploying

- Fix BranchSelector flickers when no branch is found

> - Fix broken "stop development" workflow

Reviewer 3: ```suggestion
- Fix runtime error and blank screen upon clicking "Stop development" button
```
